apps/surfer/src/Surfer.ts

```diff
@@ -0,0 +1,75 @@
+import { SurferContext, SurferRpcMessage } from './types';
+
+export interface SurferGuidelines {
+  initWithOptions: (...args: any[]) => SurferContext & {
+    $iframe: HTMLIFrameElement;
+    setPermalink: (permalink: string | null) => void;
+  };
+  setHtml: (html: string | null) => void;
+}
+
+declare global {
+  interface Window {
+    surferGuidelines: SurferGuidelines;
+  }
+}
+
+export class Surfer {
+  private iframe: HTMLIFrameElement | null = null;
+  private surferContext: SurferContext | null = null;
+  private rpcListener: ((message: MessageEvent) => void) | null = null;
+
+  constructor(private readonly surferGuidelines: SurferGuidelines, private readonly eventEmitter = window) {
+    if (!this.surferGuidelines) {
+      throw new Error('Surfer is not available');
+    }
+  }
+
+  initialize(shareToken: string, container: HTMLDivElement, onReady?: (context: SurferContext) => void) {
+    const { $iframe, setPermalink, ...surferContext } = this.surferGuidelines.initWithOptions({ partner: 'contentful' });
+
+    this.iframe = $iframe;
+    this.surferContext = surferContext;
+    setPermalink(this.buildPermalink(shareToken));
+
+    this.mount(container);
+    onReady?.(this.surferContext!);
```

Comment:
No async stuff in this function, can't we just return the context?

Reply:
Good point ✅